osa/scripts/gain_selection.py

```diff
@@ -56,21 +57,32 @@ def apply_gain_selection(date: str, output_basedir: Path = None):
     data_runs = summary_table[summary_table["run_type"] == "DATA"]
 
     output_dir = output_basedir / date
-    log_dir = output_basedir / "log"
+    log_dir = output_basedir / "log" /date
     output_dir.mkdir(parents=True, exist_ok=True)
     log_dir.mkdir(parents=True, exist_ok=True)
     r0_dir = Path(f"/fefs/aswg/data/real/R0/{date}")
 
     for run in data_runs:
         run_id = run["run_id"]
+        n_subruns = run["n_subruns"]
         ref_time = run["dragon_reference_time"]
         ref_counter = run["dragon_reference_counter"]
         module = run["dragon_reference_module_index"]
         ref_source = run["dragon_reference_source"].upper()
 
+        input_files = []
+
         if ref_source=="UCTS" or ref_source=="TIB":
 
-            input_files = r0_dir.glob(f"LST-1.1.Run{run_id:05d}.????.fits.fz")
+            for subrun in range(n_subruns):
+                new_files = glob.glob(f"{r0_dir}/LST-1.?.Run{run_id:05d}.{subrun:04d}.fits.fz")
+
+                if len(new_files) != 4:
+                    for file in new_files:
+                        shutil.copy2(file, output_dir, follow_symlinks=True)
+ 
+                else:
+                    [input_files.append(new_files[i]) for i in range(len(new_files))]
```

Comment:
Do you have to run the gain selection script separately for each of the 4 stream files of a given subrun or would it be enough to store in `input_files` the first stream file `LST-1.1.Run{run_id:05d}.{subrun:04d}.fits.fz` once you know that the 4 streams are available?

Reply:
in any case why not do simply `input_files.extend(new_files)`?

Reply:
No, in fact providing just the first stream file is enough, I will change it